### PR TITLE
Rewrite fetch source files (#249)

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #249, #248, #239 and #254.

Mainly, provides a Python version of tools/fetch_source_files.sh, for better portability and maintainability.

### CODE REVIEW

**Comment by trgiangdo** [comment]
@FabienLelaquais We need an update on the [Installation page](https://github.com/Avaiga/taipy-doc/blob/develop/INSTALLATION.md).

Besides, what do you think about explicitly explain the arguments of `tools/fetch_source_files.py` on this page?
Like provide a list of `--no_pull`, `--check`, ... with short description on the installation page so that the user know the options, and they can run `--help` for more information.

**Comment by FabienLelaquais** [comment]
> We need an update on the Installation page

Yes yes that's true. The different options are already documented in the sh and the py versions of the fetch script.
It would make a lot of sense to indicate them in the Installation md file as well, indeed.

At this stage, I have not changed the build process, which still relies on the Bash version of the script.
The plan is to move to he Python version, drop the sh version, and then update the Installation file, where your comment makes a lot of sense.

thx

File: tools/setup_generation/step_refman.py [hunk 1]
```diff
@@ -397,34 +399,31 @@ def generate_entries(
...
 
         # Read config.py file
-        with open(path, 'r') as f:
+        with open(config_path, "r") as f:
             contents = f.readlines()
 
         # Inject imports and code
-        imports_to_inject = """from types import NoneType
-from typing import Any, Callable, List
+        imports_to_inject = """from typing import Any, Callable, List
```
Comment (jrobinAV) [inline tools/setup_generation/step_refman.py:426]: you removed NoneType? 
Is it on purpose?

**Review by jrobinAV — APPROVED** [review]
On the contributing.md File I have many remarks on the form and content. But I believe it is for another ticket. Right ?

File: tools/setup_generation/step_refman.py (hunk 1 above)
Comment (FabienLelaquais) [reply tools/setup_generation/step_refman.py:426]: Well yes: it is not used... and it actually broke my setup at times...
Go figure.

**Pull request merged**